### PR TITLE
FHE: Expose ciphertext arithmetic operations

### PR DESCRIPTION
**Pull request by joeykraut**
### Purpose
This PR exposes further helper methods to support ciphertext <> ciphertext arithmetic

### CODE REVIEW

File: FHE/Ciphertext.cpp [hunk 1]
```diff
@@ -4,17 +4,39 @@
...
+  return result;
+}
 
-void Ciphertext::set(const Rq_Element& a0, const Rq_Element& a1,
-        const FHE_PK& pk)
+/**
+ * Implementation
+ */
+
+Ciphertext::Ciphertext(const FHE_PK &pk) : Ciphertext(pk.get_params())
```
Comment (akirillo) [inline FHE/Ciphertext.cpp:41]: Do we actually use this constructor anywhere? Looks like e.g. above, we just use the `Ciphertext(const FHE_Params &p)` constructor

File: FHE/FHE_Keys.cpp [hunk 2]
```diff
@@ -6,153 +6,202 @@
...
 }
 
-FHE_SK::FHE_SK(const FHE_Params& pms) :
-    FHE_SK(pms, pms.get_plaintext_modulus())
+unique_ptr<FHE_SK> get_sk(const FHE_KeyPair &pair)
 {
+  return unique_ptr<FHE_SK>(new FHE_SK(pair.sk));
 }
 
+unique_ptr<Ciphertext> encrypt(const FHE_KeyPair &pair, const Plaintext_<FFT_Data> &mess)
```
Comment (akirillo) [inline FHE/FHE_Keys.cpp]: Will we always want to encrypt/decrypt through the whole keypair, or should we also expose the underlying `pk.encrypt` / `sk.decrypt` for FFI?

**Review by akirillo — APPROVED** [review]

File: FHE/Ciphertext.cpp (hunk 1 above)
Comment (joeykraut) [reply FHE/Ciphertext.cpp:41]: yes this is used almost everywhere in the codebase

File: FHE/FHE_Keys.cpp (hunk 2 above)
Comment (joeykraut) [reply FHE/FHE_Keys.cpp]: see follow up

**Pull request merged**